### PR TITLE
Add flake

### PR DESCRIPTION
**Pull request by n8henrie**
I realize you're currently [not using flakes](https://github.com/timokau/nix-bisect/issues/23#issuecomment-1576851633), but providing flake outputs sure makes for easy no-install running of these kinds of tools, I'm sure users would appreciate the option:

```console
$ nix run github:n8henrie/nix-bisect -- --help
usage: nix-build-status [-h] [--file FILE] [--option name value] [--argstr name value] [--max-rebuilds MAX_REBUILDS]
                        [--failure-line FAILURE_LINE] [--on-success {good,bad,skip,skip-range,<int>}]
                        [--on-failure {good,bad,skip,skip-range,<int>}]
                        [--on-dependency-failure {good,bad,skip,skip-range,<int>}]
                        [--on-failure-without-line {good,bad,skip,skip-range,<int>}]
                        [--on-instantiation-failure {good,bad,skip,skip-range,<int>}]
                        [--on-resource-limit {good,bad,skip,skip-range,<int>}]
                        [--rebuild-blacklist REBUILD_BLACKLIST]
                        drvish

Build a package with nix, suitable for git-bisect.

positional arguments:
  drvish                Derivation or an attribute/expression that can be resolved to a derivation in the context of
                        the nix file

options:
  -h, --help            show this help message and exit
  --file FILE, -f FILE  Nix file that contains the attribute
  --option name value   Set the Nix configuration option `name` to `value`.
  --argstr name value   Passed on to `nix instantiate`
  --max-rebuilds MAX_REBUILDS
                        Number of builds to allow.
  --failure-line FAILURE_LINE
                        Line required in the build logs to count as a failure.
  --on-success {good,bad,skip,skip-range,<int>}
                        Bisect action if the expression can be successfully built
  --on-failure {good,bad,skip,skip-range,<int>}
                        Bisect action if the expression can be successfully built
  --on-dependency-failure {good,bad,skip,skip-range,<int>}
                        Bisect action if the expression can be successfully built
  --on-failure-without-line {good,bad,skip,skip-range,<int>}
                        Bisect action if the expression can be successfully built
  --on-instantiation-failure {good,bad,skip,skip-range,<int>}
                        Bisect action if the expression cannot be instantiated
  --on-resource-limit {good,bad,skip,skip-range,<int>}
                        Bisect action if a resource limit like rebuild count is exceeded
  --rebuild-blacklist REBUILD_BLACKLIST
                        If any derivation matching this regex needs to be rebuilt, the build is skipped
```

Of course no worries if not interested, thanks for providing the tool in the first place!

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  outputs = {
+    self,
+    nixpkgs,
+  }: let
+    inherit (nixpkgs) lib;
+    systems = ["aarch64-darwin" "x86_64-linux" "aarch64-linux"];
+    systemClosure = attrs:
+      builtins.foldl' (acc: system:
+        lib.recursiveUpdate acc (attrs system)) {}
+      systems;
```
[the remark this answers is not in the document]
Comment (n8henrie) [reply flake.nix:22]: Done! Let me know how that looks.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        pname = "nix-bisect";
+      in {
+        packages.${system} = {
+          default = self.packages.${system}.${pname};
+          ${pname} = pkgs.callPackage ./. {inherit pkgs;};
+        };
+
+        apps.${system} =
+          {
+            default = self.apps.${system}.nix-build-status;
```
[the remark this answers is not in the document]
Comment (n8henrie) [reply flake.nix]: Not necessary at all. Removed.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
```
[the remark this answers is not in the document]
Comment (n8henrie) [reply flake.nix:1]: Sure!

**Review by timokau — APPROVED** [review]
Thank you @n8henrie!

**Pull request merged**